botocore/client.py

```diff
@@ -206,8 +215,14 @@ def _get_client_args(self, service_model, region_name, is_secure,
             if client_config.user_agent_extra is not None:
                 user_agent += ' %s' % client_config.user_agent_extra
 
-        signer = RequestSigner(
-            service_name, endpoint_config['signing_region'],
+        signing_region = endpoint_config['signing_region']
+        endpoint_region_name = endpoint_config['region_name']
+        if signing_region is None and endpoint_region_name is None:
+            signing_region, endpoint_region_name = \
+                self._get_default_s3_region(service_name, endpoint_bridge)
```

Comment:
`_get_default_s3_region` can sometimes return None, but this function is always assuming a 2 element tuple is being returned.